<a href="https://colab.research.google.com/github/TeraneAgaeva/Data-Analysis-course-/blob/main/%22hw_6_2_%D0%A1%D1%82%D0%B0%D1%82%D0%B8%D1%81%D1%82%D0%B8%D1%87%D0%BD%D1%96_%D1%82%D0%B5%D1%81%D1%82%D0%B8_ipynb%22.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Домашнє завдання 6: Статистичні тести

0. Зчитайте дані з `data.csv` та запишіть значення колонки data у форматі np.ndarray в змінну data. Ми будемо далі працювати з цією змінною.

In [1]:
import numpy as np
import pandas as pd


In [2]:
import math

In [3]:
!pip install statsmodels

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [5]:
!ls drive/MyDrive/data.csv

drive/MyDrive/data.csv


In [6]:
data_path = 'drive/MyDrive/data.csv'
data = pd.read_csv(data_path)

Запустіть код нижче. Для коректної робити всіх подальших методів дані мають бути у вигляді одновимірного вектора, тобто мати розмір (100,).

In [7]:
data.shape

(100, 2)

In [8]:
data

Unnamed: 0       data
0            0  17.499453
1            1  19.662399
2            2   7.182823
3            3  29.841625
4            4   9.239386
..         ...        ...
95          95  26.244944
96          96  22.282832
97          97  18.697188
98          98  27.041189
99          99   5.938381

[100 rows x 2 columns]

In [9]:
from statsmodels.stats.weightstats import ztest

1. Зчитані дані - це сума покупок в доларах наших покупців на сайті протягом липня 2022 року.

До цього місяця, ми мали продажі в середньому на чек $20. Чи вийшло у нас статистично значущо **підвищити** середній чек за останній місяць?

Оберіть статистичний тест (z-test чи t-test). Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чеку.

**Підказка**: тут тест буде одновибірковий.

Застосовувати потрібно z-test, адже вибірка більше 30 записів

In [10]:
alpha = 0.05
mean = 20

In [11]:
ztest_Score, p_value = ztest(data.data, value = 20, alternative='larger')

ztest_Score, p_value

(-0.9953477389335053, 0.8402164165464012)

In [12]:
if(p_value <  alpha):
    print("Відхилити Н0.")
else:
    print("Н0 не може бути відхиленаб, не маємо статистично значущу різницю у чеку.")

Н0 не може бути відхиленаб, не маємо статистично значущу різницю у чеку.


2. Пройшов час, і ми зібрали дані про продажі за серпень 2022 року, які зберігли у файл `data_2.csv`. Продажів було дещо більше, але чи змінився стат. значущо середній чек порівняно з липнем 2022? Перевірте це використовуючи ztest.

  Виведіть p-значення статистичного тесту та зробіть висновок, чи ми маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями.

  **Підказка**: треба провести двовибірковий ztest з Н0 - що середні значення в липні та серпні 2022 року - однакові. Для кращого розуміння ситуації, можливо вам буде цікаво переглянути, які середні показники чеку були в кожен з місяців. Порівняйте свій висновок з простого спостереження середніх значень і висновок статистичного тесту. Чи вони збігаються?

In [13]:
!ls drive/MyDrive/data.csv

drive/MyDrive/data.csv


In [14]:
data_path_2 = 'drive/MyDrive/data_2.csv'
data_2 = pd.read_csv(data_path_2)

In [15]:
data_2

data
0     5.940159
1    29.189344
2    11.383502
3    15.434098
4    19.896752
..         ...
115  31.304916
116  15.822676
117  19.291394
118  19.036747
119  18.847899

[120 rows x 1 columns]

In [19]:
#середні показники чеку
np.mean(data.data), np.mean(data_2.data)

(19.37755319670441, 19.37322276512519)

In [17]:
ztest(data.data, data_2.data, value=0)

(0.005246673456808504, 0.9958137794593356)

In [20]:
ztest(data.data, data_2.data, alternative='two-sided')

(0.005246673456808504, 0.9958137794593356)

Статистика для двох вибірок z-критерію становить 0.0052, а відповідне p-value дорівнює 0.9958.

Оскільки p-value > 0.05, у нас НЕ достатньо доказів, щоб відкинути нульову гіпотезу. Ми НЕ маємо зараз статистично значущу різницю у розмірі середнього чека між двома місяцями. Іншими словами, середній рівень чеку змінився не значно. Це також спостерігаємо при розрахунку середнього 19,3776 та 19,3732
